libnmstate/prettystate.py

```diff
@@ -112,30 +109,24 @@ def represent_unicode(_, data):
     """
 
     return yaml.ScalarNode(
-        tag=u"tag:yaml.org,2002:str", value=data.encode("utf-8")
+        tag="tag:yaml.org,2002:str", value=data.encode("utf-8")
     )
 
 
-def order_iface_state(iface_state):
-    ordered_state = OrderedDict()
-
-    for setting in ("name", "type", "state"):
-        try:
-            ordered_state[setting] = iface_state.pop(setting)
-        except KeyError:
-            pass
-
-    for key, value in order_dict(iface_state).items():
-        ordered_state[key] = value
-
-    return ordered_state
-
+def _sort_with_priority(data):
+    if isinstance(data, Sequence) and not isinstance(data, str):
+        return [_sort_with_priority(item) for item in data]
+    elif isinstance(data, Mapping):
+        new_data = {}
+        for key in sorted(data.keys(), key=_sort_with_priority_key_func):
+            new_data[key] = _sort_with_priority(data[key])
+        return new_data
+    else:
+        return deepcopy(data)
 
-def order_dict(dict_):
-    ordered_dict = OrderedDict()
-    for key, value in sorted(dict_.items()):
-        if isinstance(value, dict):
-            value = order_dict(value)
-        ordered_dict[key] = value
 
-    return ordered_dict
+def _sort_with_priority_key_func(key):
+    try:
+        return f"{PRIORITY_LIST.index(key)}"
+    except ValueError:
+        return key
```

Comment:
This seems to fail when trying to sort

```
example = {"0":1,
           "name": "example"
          }
```

This should fix it:
```suggestion
def _prioritize(key):
    try:
        priority = PRIORITY_LIST.index(key)
    except ValueError:
        priority = len(PRIORITY_LIST)
    return (priority, key)
```

Reply:
Thanks. Patch applied.